### PR TITLE
feat: Extending NetworkAnnotation class to enable DPDK support in UPF

### PR DESCRIPTION
**Pull request by Gmerold**
# Description

This PR adds additional fields to the `NetworkAnnotation` dataclass:
- mac
- ips
New fields are required to create UPF network annotations when running UPF in DPDK mode. 

# Checklist:

- [ ] My code follows the [style guidelines](/CONTRIBUTING.md) of this project
- [ ] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that validate the behaviour of the software
- [ ] I validated that new and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules
- [ ] I have bumped the version of the library

### CODE REVIEW

**Review by gatici — APPROVED** [review]

File: requirements.txt [hunk 1]
```diff
@@ -1,3 +1,4 @@
+dataclass-wizard
```
Comment (gruyaume) [inline requirements.txt:1]: I'm not convinced about adding this requirement. This project is barely maintained and only has 2 contributors.

**Review by gruyaume — CHANGES_REQUESTED** [review]

File: requirements.txt (hunk 1 above)
Comment (Gmerold) [reply requirements.txt]: Yeah, true, I didn't check that.
I've replaced the lib with custom implementation of the `dict` method.

**Review by ghislainbourgeois — APPROVED** [review]

File: lib/charms/kubernetes_charm_libraries/v0/multus.py [hunk 2]
```diff
@@ -150,8 +150,16 @@ class NetworkAnnotation:
...
+    ips: Optional[List[str]] = None
 
-    dict = asdict
+    def dict(self) -> dict:
+        """Returns a NetworkAnnotation in the form of a dictionary.
+
+        Returns:
+            dict: Dictionary representation of the NetworkAnnotation
+        """
+        return {key: value for key, value in asdict(self).items() if value}
```
Comment (gruyaume) [inline lib/charms/kubernetes_charm_libraries/v0/multus.py:162]: Why this change?

Comment (Gmerold) [reply lib/charms/kubernetes_charm_libraries/v0/multus.py:162]: That's because we don't want `None`s in the dict. If we had `None`s, then the StatefulSet annotation could look something like:
```
"k8s.v1.cni.cncf.io/networks": {"name": "what", "interface": "ever", "mac": null, "ips": null}
```
I figured it'd be better to filter out `None`s and keep the annotation clean.

**Review by gruyaume — APPROVED** [review]
